src/libsyntax/parse/token.rs

```diff
@@ -323,6 +323,22 @@ impl Token {
         }
     }
 
+    /// Returns `true` if token is special identifier in module name
+    #[allow(non_upper_case_globals)]
+    pub fn is_special_ident_in_modname(&self) -> bool {
+        match *self {
+            Ident(sid, ModName) => {
+                let n = sid.name;
+
+                   n == SELF_KEYWORD_NAME
+                || n == STATIC_KEYWORD_NAME
+                || n == SUPER_KEYWORD_NAME
+                || n == SELF_TYPE_KEYWORD_NAME
```

Comment:
Sure. But than name of function will be incorrect. Is it ok to rename to `is_self_or_super_in_modname`.
Or move this checking to parser? In first commit this function was meaningful, now it's function for one call.